test/test_dependency_versions.py

```diff
@@ -0,0 +1,142 @@
+    if not build_tools_path.is_dir():
+        raise RuntimeError(
+            "$SCHRODINGER_SRC/mmshare/build_tools cannot be found.")
+    sys.path.insert(0, str(build_tools_path))
+    try:
+        import package_factory
+    except Exception as err:
+        raise RuntimeError("The package factory module could not be imported "
+                           f"from {build_tools_path}:\n"
+                           f"\t{str(err)}")
+    finally:
+        sys.path.pop(0)
+    return package_factory
+
+
+package_factory = import_package_factory()
+
+
+def get_dependency_versions_from_mmshare(
+        packages: Iterable) -> dict[str, tuple[int]]:
+    """
+    Get information about package versions from mmshare.
+
+    :return: Version information as a dictionary of
+      {package name: (major version, minor version, patch version)}
+    """
+    logger = logging.getLogger("test_dependency_versions")
+    mmshare_pf = package_factory.Pixi(logger, MMSHARE_PF_ENVIRONMENT)
+    mmshare_manifest_file = package_factory.Pixi.get_requirements_file()
+    # Build a pattern that matches only the packages we're looking for. The
+    # carat and dollar sign specify that the package name must be an exact match
+    # without any extra preceding or trailing characters, since we don't want to
+    # match, e.g., xgboost when looking for boost.
+    packages_description = "|".join(f"^{dep}$" for dep in packages)
+    # --no-lockfile-update tells Pixi not to make any changes to the mmshare
+    # build env, even if it's out of date or not built yet
+    cmd = [
+        "list", packages_description, "--manifest-path",
+        str(mmshare_manifest_file), "-e", MMSHARE_PF_ENVIRONMENT,
+        "--no-lockfile-update", "--json"
+    ]
+    json_data = mmshare_pf.run(cmd, capture_output=True)
+    dep_info = json.loads(json_data)
+    return {
+        cur_dep["name"]: normalize_version(cur_dep['version'])
+        for cur_dep in dep_info
+    }
+
+
+def normalize_version(version_text: str) -> tuple[int]:
+    """
+    Normalize the version strings that have been retrieved from package factory
+    and convert everything to tuples of ints. For example, this method will
+    convert
+      - 1_81_0_py311_cpp20
+      - 3.4.0_e67c494c_bldmgr_9082
+      - v1.5.5_boost_1_81_0_BLDMGR_7247
+      - release_2024_9_6_bldmgr_9464
+    to (1, 81, 0), (3, 4, 0), (1, 5, 5), and (2024, 9, 6)
+    """
+    # remove any leading "v" or "release" prefixes
+    version_text = version_text.removeprefix("v")
+    version_text = version_text.removeprefix("release")
+    version_text = version_text.lstrip("_")
+    # ignore any fields that contains letters, as well as anything afterwards
+    version_fields = re.split(r"[._]", version_text)
+    numeric_fields = itertools.takewhile(lambda field: field.isdigit(),
+                                         version_fields)
+    return tuple(map(int, numeric_fields))
+
+
+def test_dependency_versions():
```

Comment:
maybe this is silly -- could you load sketcher_versions into a global and parameterize the test on those versions, and then xfail the packages in PACKAGES_TO_IGNORE? (Maybe not empscripten, but I kinda want a report when running this that Qt, boost, and fmt are out of sync)